modules/oe_webtools_etrans/src/Plugin/Block/UnifiedEtransBlock.php

```diff
@@ -0,0 +1,309 @@
+    $translation_exists = (bool) $this->getRouteEntityLangcode();
+    return AccessResult::allowedIf(!$translation_exists);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), [
+      'url.path',
+      'languages:' . LanguageInterface::TYPE_INTERFACE,
+    ]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $placeholder_id = Html::getUniqueId('etrans-widget');
+    // On node route, we take the node original language,
+    // which is not necessarily the default language of the website.
+    $translation_from = $this->getRouteEntityLangcode(TRUE) ?: $this->languageManager->getDefaultLanguage()->getId();
+    $translation_to_language = $this->languageManager->getCurrentLanguage();
+
+    $json = $this->preparesWtEtransJson($translation_from, $placeholder_id);
+
+    // Returns UEC webtool eTrans widget.
+    $build['etrans_uec'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'script',
+      '#value' => Json::encode($json),
+      '#attributes' => [
+        'type' => 'application/json',
+      ],
+      '#attached' => [
+        'library' => [
+          'oe_webtools_etrans/wt_etrans',
+        ],
+        'drupalSettings' => [
+          'path' => [
+            'languageTo' => $translation_to_language->getId(),
+          ],
+        ],
+      ],
+    ];
+
+    $build['widget_placholder'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'id' => $placeholder_id,
+      ],
+    ];
+
+    $message = $this->t("@language is available via eTranslation, the European Commission's machine translation service.", [
+      '@language' => $translation_to_language->getName(),
+    ], [
+      'langcode' => $translation_to_language->getId(),
+    ]);
+    $translate_link = [
+      '#type' => 'link',
+      '#url' => Url::fromRoute('<none>', [], ['attributes' => ['class' => ['oe-webtools-unified-etrans--translate']]]),
+      '#title' => $this->t('Translate to @language', [
+        '@language' => $translation_to_language->getName(),
+      ], [
+        'langcode' => $translation_to_language->getId(),
+      ]),
+    ];
+    $disclaimer_link = [
+      '#type' => 'link',
+      '#url' => Url::fromUri('https://commission.europa.eu/languages-our-websites/use-machine-translation-europa_' . $translation_to_language->getId(), [
+        'attributes' => [
+          'class' => ['webtools-etrans--disclaimer'],
+          'target' => '_blank',
+        ],
+      ]),
+      '#title' => $this->t('Important information about machine translation', [], ['langcode' => $translation_to_language->getId()]),
+    ];
+    $build['translation_request'] = [
+      '#theme' => 'block__unified_etrans_request',
+      '#message' => $message,
+      '#translate_link' => $translate_link,
+      '#disclaimer_link' => $disclaimer_link,
+    ];
+
+    return $build;
```

Comment:
I think the translation_to_language should be added as a cacheable dependency.

Reply:
Well, you need the block to work for anonymous user. The language is selected from the language switcher. As cache is really hard to implement for anonymous user, I think it's an overkill and really hard to implement to actually cache the $translate_to_language var in the block.